io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/AbstractTableInstanceWriter.java

```diff
@@ -225,20 +234,40 @@ private boolean shouldBeDisplayed(Object obj) {
 	}
 
 	private void addProperty(List<String> headerRow, Map<String, Object> row, Object property,
-			String propertyTypeName) {
+			String propertyTypeName, IOReporter reporter) {
 		if (!headerRow.contains(propertyTypeName)) {
 			headerRow.add(propertyTypeName);
 		}
+
+		if (property instanceof DefaultGeometryProperty<?>) {
+			// Property is an instance of DefaultGeometryProperty<T>
+			DefaultGeometryProperty<?> geometryProperty = (DefaultGeometryProperty<?>) property;
+
+			Geometry geometry = geometryProperty.getGeometry();
+
+			// correct winding order as per right-hand rule, i.e.,
+			// exterior rings are counterclockwise, and holes are
+			// clockwise.
+			setWindingOrder(EnumWindingOrderTypes.counterClockwise);
+			property = unifyGeometry(geometry, reporter,
+					geometryProperty.getCRSDefinition().getCRS());
```

Comment:
I don't think for CSV/Excel there is a rule on what the winding order should be. Since you extended `AbstractGeoInstanceWriter` it would make sense to use its configuration parameters.
In that case

- you should use `extractGeometry` to detect and determine geometry and CRS, and apply conversion and the winding order
- extend the writer definitions (in the respective extension) with the additional parameters (target CRS, winding order - not sure about the EPSG prefix - is the CRS information included in the resulting file?)
- extend the UI to make the additional parameters configurable

Reply:
@stempler please check the new commit. 
I believe that we should avoid using `extractGeometry` and `setting winding order`. Instead, when writing the property, we should simply check if the geometry is a multi-geometry. If it contains only one geometry, we should write that single geometry. Otherwise, we should follow the existing procedure.
Maybe the aproach that you are describing to create a UI to ask the user the winding order could be done as part of another ticker and extended to all the data export formats. 
In the [InstanceToJson](https://github.com/halestudio/hale/blob/247bc7ce6f33207783639bd178249cfd1413c455/io/plugins/eu.esdihumboldt.hale.io.json/src/eu/esdihumboldt/hale/io/json/internal/InstanceToJson.java#L652) I've see that the  `windingOrder = true` always.